bridge_adaptivity/bridge_lti/consumer.py

```diff
@@ -36,9 +36,15 @@ def source_preview(request):
     """View to render Source content block shared through LTI."""
     log.debug("Got request.GET: %s", request.GET)
 
-    content_provider = get_content_provider()
-    if not content_provider:
+    content_source_id = request.GET.get('content_source_id')
+    content_providers = get_content_providers(content_source_id)
+    require_content_source_id = content_providers.count() > 1
+
+    if require_content_source_id and not content_source_id:
         return render(request, 'bridge_lti/stub.html')
+
+    content_provider = content_providers.first()
```

Comment:
It seems this part is not finished yet. Why we take the first of content_providers what about others?

Reply:
I already explained it to you. 
because here we will have only one in queryset.  

Probably I should better to add comment here.